src/main/java/duke/util/Parser.java

```diff
@@ -99,14 +100,47 @@ public static int filterTaskIndex(String command) throws MissingIndexException {
             throw new MissingIndexException();
         }
 
-        String indexString = command.split(" ", 2)[1];
+        String indexString = commandItems[1];
         if (indexString.trim().isEmpty()) {
             throw new MissingIndexException();
         }
 
         return parseInt(indexString);
     }
 
+    /**
+     * Takes in the original inputted command and filters and parses it for the given integer index.
+     * The index is the full command string minus the first word which is the command itself.
+     *
+     * @param command the original command inputted by the user.
+     * @return the filtered and parsed int that refers to the index of the task.
+     * @throws MissingIndexException when index is missing (including if it contains only white space).
+     */
+    public static int[] filterMultipleTaskIndexes(String command) throws MissingIndexException {
```

Comment:
Perhaps a more apt function name could be getAllTaskIndexes?

Reply:
Thanks for the suggestion! I think that it probably a better way of naming the function than my original name.